---
title: "Ratio distribution plots and Scatter plots"
author: "Xiaowen Shi, Chen Chen"
date: "09/18/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 1. Ratio distribution plots
A random RNA-Seq dataset was generated for the demonstration, composed of 3 biological replicates for the experimental group and the control group. Counts for 300 cis genes and 1700 trans genes are shown.
```{r}
set.seed(2018)
cis_expect_peak = 1.2
trans_expect_peak = 0.99
count_mean = 200
count_sd = 100
num_trans = 1700
num_cis = 300

expect_reads <- round(abs(rnorm(n = num_trans+num_cis,mean = count_mean,sd = count_sd)))
expect_reads_treatment <- round(abs(c(expect_reads[1:num_cis]*rnorm(num_cis,cis_expect_peak,cis_expect_peak/4),expect_reads[(num_cis+1):length(expect_reads)]*rnorm(num_trans,trans_expect_peak,trans_expect_peak/4))))
counts <- NULL
for (i in 1:3) {
  counts <- cbind(counts,rpois(n =  num_trans+num_cis,lambda = expect_reads))
}
for (i in 1:3) {
  counts <- cbind(counts,rpois(n =  num_trans+num_cis,lambda = expect_reads_treatment))
}
colnames(counts) <- c('c1','c2','c3','t1','t2','t3') #3 controls and 3 treatments
cis_genes <- paste0('cis_gene',1:num_cis)
trans_genes <- paste0('trans_gene',1:num_trans)
rownames(counts) <- c(cis_genes,trans_genes) 
gene_length <- abs(round(rnorm(n = 2000,mean = 800,sd = 400)))
head(counts)
```

a.Normalization of read counts(with rpkm() in edgeR).
```{r}
library(edgeR)
source('plot_utils.R')
rpkm_data <- rpkm(counts,gene.length=gene_length)
head(rpkm_data)
```

b.Remove lowly expressed genes.
```{r}
rpkm_data_filtered <- rpkm_data[apply(rpkm_data, 1, function(x)sum(x>0))>3,]
head(rpkm_data_filtered)
```

c.Compute the mean of normalized counts.
```{r}
mean_control <- apply(rpkm_data_filtered[,1:3], 1, mean)
mean_treatment <- apply(rpkm_data_filtered[,4:6], 1, mean)
mean_control[mean_control==0] <- 10e-6
```

Calculate the ratio of each gene.
```{r}
r <- mean_treatment/mean_control
```

d.Generate a histogram using ggplot2 package.
```{r,fig.width=5.7, fig.height=4.27 }
distribution_plot <- plot_distribution(r,title_name = '',left_line = 0.67,right_line = 1.5)
plot(distribution_plot)
```

e.Plot cis and trans genes seperately.
```{r,fig.width=5.7, fig.height=4.27}
distribution_plot2 <- plot_distribution_pairs(r,cis_genes,title_name = '',left_line = 0.67,right_line = 1.5,max_ratio = 6)
```

# 2.Scatter plots
a.Perform differential gene expression analysis using edgeR.

```{r}
group_table <- data.frame('group'=rep('experiment1',6),
                          'condition'=c(rep('control',3),rep('treatment',3)),
                          'control'=c(rep(T,3),rep(F,3)))
rownames(group_table) <- colnames(counts)
de <- edgeR_wrapper(cnt = counts,grp_table = group_table)
de1 <- de$experiment1.treatment
head(de1)
```

b.Compute mean of normalized counts.
```{r}
avg_expre <- apply(rpkm_data, 1, mean)
```

c.Generate a scatter plot.
```{r,fig.width=5.7, fig.height=4.27}
plot_scatter(Fold_Change = 2^de1$logFC,Expre = avg_expre,P_Value = de1$FDR,left_line = 0.67,right_line = 1.5)
```

d.Plot cis and trans genes seperately.
```{r,fig.width=5.7, fig.height=4.27*1.5}
fc <- 2^de1$logFC
pval <- de1$FDR
names(fc) <- names(pval) <- names(avg_expre)
plot_scatter_pairs(Fold_Change = fc,Expre = avg_expre,P_Value = pval,cis_genes,left_line = 0.67,right_line = 1.5)
```
